tests/auxiliary/minicore.rs

```diff
@@ -0,0 +1,116 @@
+//! Auxiliary `minicore` prelude which stubs out std preludes for tests that need to work in
+//! cross-compilation scenarios where no `std` is available (that don't need to `-Zbuild-std`).
+
+#![feature(no_core, lang_items, rustc_attrs, unsized_fn_params, transparent_unions)]
+#![allow(unused, improper_ctypes_definitions, internal_features)]
+#![no_std]
+#![no_core]
+
+#[lang = "sized"]
+pub trait Sized {}
+
+#[lang = "receiver"]
+pub trait Receiver {}
+impl<T: ?Sized> Receiver for &T {}
+impl<T: ?Sized> Receiver for &mut T {}
+
+#[lang = "copy"]
+pub trait Copy: Sized {}
+impl Copy for i32 {}
+impl Copy for f32 {}
+impl<T: ?Sized> Copy for &T {}
+impl<T: ?Sized> Copy for *const T {}
+impl<T: ?Sized> Copy for *mut T {}
+
+#[lang = "clone"]
+pub trait Clone: Sized {
+    fn clone(&self) -> Self;
+}
+
+#[lang = "phantom_data"]
+pub struct PhantomData<T: ?Sized>;
+impl<T: ?Sized> Copy for PhantomData<T> {}
+
+#[lang = "unsafe_cell"]
+#[repr(transparent)]
+pub struct UnsafeCell<T: ?Sized> {
+    value: T,
+}
+
+pub trait Any: 'static {}
+
+pub enum Option<T> {
+    None,
+    Some(T),
+}
+impl<T: Copy> Copy for Option<T> {}
+
+pub enum Result<T, E> {
+    Ok(T),
+    Err(E),
+}
+impl<T: Copy, E: Copy> Copy for Result<T, E> {}
+
+#[lang = "manually_drop"]
+#[repr(transparent)]
+pub struct ManuallyDrop<T: ?Sized> {
+    value: T,
+}
+impl<T: Copy + ?Sized> Copy for ManuallyDrop<T> {}
```

Comment:
You can derive `Copy` if you add

```rust
#[rustc_builtin_macro]
pub macro Copy($item:item) {
    /* compiler built-in */
}
```

and equivalent for `Clone`.

Reply:
I did not make this change because I couldn't figure out how to make use of this, just kept manual `impl Copy for ...` for now.